### ISSUE TITLE
Registering multiple custom Problems inside a Flask API fails

### ISSUE THREAD

**Issue by mabergerx** [tensorflow/tensor2tensor#1066]
### Description

[Asked a question about it on SO:](https://stackoverflow.com/questions/52315125/importing-multiple-custom-tensor2tensor-problems-in-flask) but figured that maybe the problem is deeper (or more shallow ;)) than Flask. My problem arises when I'm trying to register multiple custom Problems from multiple t2t_usr_dirs with this function:

```python
def get_registered_problem(usr_dir_string, problem):
    usr_dir.import_usr_dir(usr_dir_string)
    print("This worked?")
    problem = registry.problem(problem)
    return problem
```
For a test, I tried to do just this at the initialization type of my flask-restplus API:

```python
if __name__ == '__main__':
    problem_1 = get_registered_problem(configuration['problems']['problem1']['usr_dir'], configuration['problems']['problem1']['problem_name'])
    print(problem_1)
    problem_2 = get_registered_problem(configuration['problems']['problem2']['usr_dir'], configuration['problems']['problem2']['problem_name'])
    print(problem_2)
    app.run(host='0.0.0.0', port=6010, debug=True)
```

When just doing the above for a single Problem, everything works good. The Problem gets imported, registered and can be used. However, when trying this for two Problems, the first one goes fine again but the second one gives this error on the `problem = registry.problem(problem)` line (so it gets past the usr_dir import):

`LookupError: problem2 not in the set of supported problems:`

Could this be a Flask(-restplus) problem or is it that I can't import and register two custom Problems in one program?

Thank you very much in advance for thinking with me!

**Comment by mabergerx** [comment]
I've narrowed the behaviour to what seems like a problem with importing multiple usr_dirs with `usr_dir.import_usr_dir()`. When importing just one usr dir everything is perfect, while importing two of them the other one doesn't seem to get picked up:

```python
usr_dir.import_usr_dir(configuration['problems']['problem1']['usr_dir'])
problem1 = registry.problem(configuration['problems']['problem1']['problem_name'])
print(problem1)

usr_dir.import_usr_dir(configuration['problems']['problem2']['usr_dir'])
problem2 = registry.problem(configuration['problems']['problem2']['problem_name'])
print(problem2)
```

produces the following output:

```
INFO:tensorflow:Importing user module trainer from path my/path/to/problem1
<trainer.problem.Problem1 object at 0x7f09a691bef0>
INFO:tensorflow:Importing user module trainer from path my/path/to/problem2
Traceback (most recent call last):
  File "run_service.py", line 89, in <module>
    problem2 = registry.problem(configuration['problems']['problem2']['problem_name'])
  File "/home/datalab/anaconda3/lib/python3.6/site-packages/tensor2tensor/utils/registry.py", line 277, in problem
    raise LookupError(error_msg)
LookupError: problem2 not in the set of supported problems
```

Any ideas?

**Comment by mabergerx** [comment]
To come back to this, I found a solution for my case. For everyone who would stumble upon this in the future:

I found out that using the `usr_dir.import_usr_dir(usr_dir_string)` is not the best way to go. I solved this problem by adding the corresponding usr_dirs into my root folder and just importing them as a module like this `from my_problem1.trainer.problem import *`

After that I removed the `usr_dir.import_usr_dir(usr_dir_string)` line at all and proceded to register the problem via `problem = registry.problem(problem)`, which went fine for both problems!
